### PR TITLE
Home work 27

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/src/main/java/controller/NewUserServlet.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.UUID;
+
+@WebServlet(name = "NewUserServlet", value = "/admin/users/add")
+public class NewUserServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    private static long id = IdGenerator.generateId();
```
Comment (Sofasmile) [inline src/src/main/java/controller/NewUserServlet.java]:
```suggestion
    private static Long id = IdGenerator.generateId();
```

File: src/src/main/java/controller/ProductRegistrationServlet.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/new_product")
+public class ProductRegistrationServlet extends HttpServlet {
+
+    private static final ProductService PRODUCT_SERVICE = ProductServiceFactory.getInstance();
+    private static long id = IdGenerator.generateId();
```
Comment (Sofasmile) [inline src/src/main/java/controller/ProductRegistrationServlet.java]: the same

File: src/src/main/java/controller/ProductRegistrationServlet.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/new_product")
+public class ProductRegistrationServlet extends HttpServlet {
+
+    private static final ProductService PRODUCT_SERVICE = ProductServiceFactory.getInstance();
```
Comment (Sofasmile) [inline src/src/main/java/controller/ProductRegistrationServlet.java]:
```suggestion
    private static final ProductService productService = ProductServiceFactory.getInstance();
```

File: src/src/main/java/controller/filter/UserAccessFilter.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package controller.filter;
+
+import controller.ChangeProductServlet;
+import model.User;
+import org.apache.log4j.Logger;
+
+import javax.servlet.*;
```
Comment (Sofasmile) [inline src/src/main/java/controller/filter/UserAccessFilter.java:7]: :)

File: src/src/main/java/dao/impl/ProductDAOImpl.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    public List<Product> getAll() {
+        return DataBase.products;
+    }
+
+    @Override
+    public Optional<Product> getProduct(long productID) {
+        return DataBase.products.stream()
+                .filter(e -> e.getId() == (productID))
```
Comment (Sofasmile) [inline src/src/main/java/dao/impl/ProductDAOImpl.java]: better use equals

File: src/src/main/java/dao/impl/UsersDAOImpl.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override
+    public List getAll() {
+        return DataBase.users;
+    }
+
+    @Override
+    public Optional<User> getUserById(long userId) {
+        return DataBase.users.stream()
+                .filter(e -> e.getId() == (userId))
```
Comment (Sofasmile) [inline src/src/main/java/dao/impl/UsersDAOImpl.java]: the same

File: src/src/main/java/storage/DataBase.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataBase {
+
+    public static List<Product> products = new ArrayList<>();
+    public static List<User> users = new ArrayList<>();
+    public static List<Basket> baskets = new ArrayList<>();
+    public static List<OrderDetails> orders = new ArrayList<>();
```
Comment (Sofasmile) [inline src/src/main/java/storage/DataBase.java:16]: where is Basket and OrderDetails classes?

Comment (UnitOmega13) [reply src/src/main/java/storage/DataBase.java:16]: everything is fixed according to comments

File: src/src/main/java/model/Basket.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
+package model;
+
+import java.util.*;
```
Comment (Sofasmile) [inline src/src/main/java/model/Basket.java:3]: :) in all classes

File: src/src/main/java/model/Basket.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+public class Basket {
+    private User user;
+    private List<Optional<Product>> products = new ArrayList<Optional<Product>>();
```
Comment (Sofasmile) [inline src/src/main/java/model/Basket.java:7]: do we need optional here?

File: src/src/main/java/service/Impl/UserServiceInpl.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
+package service.Impl;
+
+import dao.UsersDAO;
+import factories.UserDAOFactory;
+import model.User;
+import service.UserService;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
```
Comment (Sofasmile) [inline src/src/main/java/service/Impl/UserServiceInpl.java:10]: do we use this library?

File: src/src/main/java/service/Impl/BasketServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public void createBasket(User user) {
+        basketDAO.createBasket(user);
+    }
+
+    @Override
+    public Optional<Basket> getUserBasket(User user) {
+        return getAllBaskets().stream()
```
Comment (Sofasmile) [inline src/src/main/java/service/Impl/BasketServiceImpl.java:28]: we meed to move this logic in dao layer
the same with all methods in this class

Comment (UnitOmega13) [reply src/src/main/java/service/Impl/BasketServiceImpl.java:28]: fixed, but unfortunately, I pushed it in another branch, I changed the link in the table

**Pull request closed without merging**